Meetme.ProfileService/Meetme.ProfileService.Tests/Services/ProfileServiceTests.cs

```diff
@@ -0,0 +1,260 @@
+﻿using AutoFixture;
+using MapsterMapper;
+using Meetme.ProfileService.BLL.Exceptions;
+using Meetme.ProfileService.BLL.Models.ProfileModels;
+using Meetme.ProfileService.DAL.Entities;
+using Meetme.ProfileService.DAL.Repositories.Interfaces;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using System.Linq.Expressions;
+
+namespace Meetme.ProfileService.Tests.Services;
+
+public class ProfileServiceTests
+{
+	private readonly BLL.Services.ProfileService _sut;
+	private readonly IRepository<ProfileEntity> _profileRepositoryMock;
+	private readonly IMapper _mapperMock;
+	private readonly IFixture _fixture;
+
+	public ProfileServiceTests()
+	{
+		_fixture = new Fixture();
+
+		_fixture.Behaviors.Remove(new ThrowingRecursionBehavior());
+		_fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+
+		_profileRepositoryMock = Substitute.For<IRepository<ProfileEntity>>();
+		_mapperMock = Substitute.For<IMapper>();
+
+		_sut = new BLL.Services.ProfileService(_profileRepositoryMock, _mapperMock);
+	}
+
+	[Fact]
+	public async Task AddAsync_ShouldAddProfile_WhenIdentityIdIsUnique()
+	{
+		//Arrange
+		var cancellationToken = _fixture.Create<CancellationToken>();
+
+		var profileEntity = _fixture.Create<ProfileEntity>();
+		var createProfileModel = _fixture.Create<CreateProfileModel>();
+
+		_profileRepositoryMock.GetFirstOrDefaultAsync(Arg.Any<Expression<Func<ProfileEntity, bool>>>(), cancellationToken).ReturnsNull();
+
+		_mapperMock.Map<ProfileEntity>(createProfileModel).Returns(profileEntity);
+
+		//Act
+		await _sut.AddAsync(createProfileModel, cancellationToken);
+
+		//Assert
+		await _profileRepositoryMock.Received(1).GetFirstOrDefaultAsync(Arg.Any<Expression<Func<ProfileEntity, bool>>>(), cancellationToken);
+		await _profileRepositoryMock.Received(1).AddAsync(profileEntity, cancellationToken);
+		
+		_mapperMock.Received(1).Map<ProfileEntity>(createProfileModel);
+	}
+
+	[Fact]
+	public async Task AddAsync_ShouldThrowBusinessLogicException_WhenIdentityIdIsNotUnique()
+	{
+		//Arrange
+		var cancellationToken = _fixture.Create<CancellationToken>();
+
+		var profileEntity = _fixture.Create<ProfileEntity>();
+		var identityProfile = _fixture.Create<ProfileEntity?>();
+		var createProfileModel = _fixture.Create<CreateProfileModel>();
+
+		_profileRepositoryMock
+			.GetFirstOrDefaultAsync(Arg.Any<Expression<Func<ProfileEntity, bool>>>(), cancellationToken)
+			.Returns(identityProfile);
+
+		_mapperMock.Map<ProfileEntity>(createProfileModel).Returns(profileEntity);
+
+		//Act
+		async Task action() => await _sut.AddAsync(createProfileModel, cancellationToken);
+
+		//Assert
+		var exception = await Assert.ThrowsAsync<BusinessLogicException>(action);
+		Assert.Equal("Profile for this identity already exists", exception.Message);
```

Comment:
use Shouldly for asserts
don't assert any hardcoded messages in tests,there is no need to test them